implementations/rust/vault/src/software/mod.rs

```diff
@@ -41,17 +63,20 @@ impl Default for DefaultVault {
 }
 
 impl DefaultVault {
+    fn cast_secret(context: &Box<dyn Secret>) -> Result<usize, VaultFailError> {
```

Comment:
This feels like an unwrap. Maybe that's a better name for it.

Reply:
I think it should be called "downcast_secret"